internal/matrix/matrix_test.go

```diff
@@ -59,69 +59,53 @@ func TestConvertMaptoInts(t *testing.T) {
 	}
 }
 
-func TestClientErrorHandling(t *testing.T) {
+func TestGenerateFibMatrix(t *testing.T) {
 
 	data := []struct {
 		name     string
 		rows     int
 		columns  int
-		expected error
+		expected [][]int64
+		err		 error
 	}{
 		{
 			name:     "Invalid Range",
 			rows:     0,
 			columns:  7,
-			expected: errors.New("You must enter a number greater than 0 and less than 93"),
+			expected: nil,
+			err: errors.New("You must enter a number greater than 0 and less than 93"),
 		},
 		{
 			name:     "Invalid Total Size",
 			rows:     10,
 			columns:  11,
-			expected: errors.New("Can only support a matrix with maximum of 92 numbers"),
-		},
-		{
-			name:     "Valid Client Inputs",
-			rows:     3,
-			columns:  7,
 			expected: nil,
+			err: errors.New("Can only support a matrix with maximum of 92 numbers"),
 		},
-	}
-
-	for _, d := range data {
-		t.Run(d.name, func(t *testing.T) {
-			result := ClientErrorHandling(d.rows, d.columns)
-			if result == nil && d.expected != nil {
-				t.Errorf("Expected %d, got %d", d.expected, result)
-			} else if result != nil && d.expected == nil {
-				t.Errorf("Expected %d, got %d", d.expected, result)
-			} else if result != nil && d.expected != nil {
-				if result.Error() != d.expected.Error() {
-					t.Errorf("Expected %d, got %d", d.expected, result)
-				}
-			}
-		})
-	}
-}
-
-func TestFibonacciMatrix(t *testing.T) {
-
-	data := []struct {
-		name     string
-		m        int
-		n        int
-		expected [][]int64
-	}{
 		{
 			name:     "Valid Matrix",
-			m:        3, // num of lists
-			n:        4, // num of elements in list
+			rows:     3,
+			columns:  4,
 			expected: [][]int64{{0, 1, 1, 2}, {34, 55, 89, 3}, {21, 13, 8, 5}},
+			err:	  nil,
 		},
 	}
 
 	for _, d := range data {
 		t.Run(d.name, func(t *testing.T) {
-			result := FibonacciMatrix(d.m, d.n)
+			result, err := GenerateFibMatrix(d.rows, d.columns)
+			
+			// Compares errors
+			if err == nil && d.err != nil {
+				t.Errorf("Expected %d, got %d", d.err, err)
+			} else if err != nil && d.err == nil {
+				t.Errorf("Expected %d, got %d", d.err, err)
+			} else if err != nil && d.err != nil {
+				if err.Error() != d.err.Error() {
+					t.Errorf("Expected %d, got %d", d.err, err)
+				}
+			}
```

Comment:
As advised in a different PR, try to use https://pkg.go.dev/github.com/stretchr/testify/require for test expectations